Source/Scene/ModelExperimental/RenderResources.js

```diff
@@ -0,0 +1,235 @@
+import defined from "../../Core/defined.js";
+import ShaderBuilder from "../../Renderer/ShaderBuilder.js";
+import DepthFunction from "../DepthFunction.js";
+import ModelExperimentalUtility from "./ModelExperimentalUtility.js";
+
+/**
+ * Resources assigned at each level of a 3D model (model, node, mesh primitive)
+ * that are required to generate {@link DrawCommand}s. These resources are passed
+ * through the various model pipeline stages (see for example
+ * {@link GeometryPipelineStage}) and updated in place. Finally,
+ * {@link buildDrawCommand} is called to construct the draw command.
+ *
+ * @private
+ */
+var RenderResources = {};
```

Comment:
I plan to add some unit tests for these classes to make sure fields are propagated correctly, especially when inheriting fields